src/app/CommandSender.h

```diff
@@ -375,6 +376,26 @@ class CommandSender final : public Messaging::ExchangeDelegate
 
     TLV::TLVWriter * GetCommandDataIBTLVWriter();
 
+    /**
+     * API for adding request data.  The `aEncodable` is generally expected to encode
+     * a ClusterName::Commands::CommandName::Type struct, however any object should work.
+     *
+     * @param [in] aCommandPath  The path of the command being requested.
+     * @param [in] aEncodable - an encodable that places the command data structure
+     *             for `aResponseCommandId` into a TLV Writer.
+     * @param [in] aAddRequestDataParams parameters associated with building the
+     *             InvokeRequestMessage that are associated with this request.
+     *
+     * This API does not validate if the command provided requires interaction use a timed
+     * invoke interaction. If the caller wants that certainty they should call the templated
+     * version of AddRequestData.
+     */
+    CHIP_ERROR AddRequestData(const CommandPathParams & aCommandPath, const DataModel::EncodableToTLV & aEncodable,
+                              AddRequestDataParameters & aAddRequestDataParams)
+    {
+        return AddRequestDataInternal(aCommandPath, aEncodable, aAddRequestDataParams);
```

Comment:
Why do we need the separate AddRequestDataInternal() function at all?  Just implement `AddRequestData` out of line with these parameters, and call it from the other AddRequestData.